realm/src/main/java/io/realm/Realm.java

```diff
@@ -1138,6 +1372,7 @@ boolean contains(Class<?> clazz) {
      * @return A sorted RealmResults containing the objects.
      * @throws java.lang.IllegalArgumentException if a field name does not exist.
      */
+    @SuppressWarnings("unchecked")
```

Comment:
If these methods are tested they should now appear as unchecked. Remove warning suppressor or implement unit test